homeassistant/components/ecobee/climate.py

```diff
@@ -270,6 +274,12 @@ def resume_program_set_service(service):
         "set_dst_mode",
     )
 
+    platform.async_register_entity_service(
+        SERVICE_SET_HUMIDIFIER_MODE,
+        {vol.Optional(ATTR_HUMIDIFIER_MODE): cv.string},
```

Comment:
It doesn't appear that set_humidifier_mode is set to deal with the mode being optional. Should this be required?

Reply:
yes, changed.